core/src/trezor/wire/message_handler.py

```diff
@@ -0,0 +1,234 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from storage.cache_common import InvalidSessionError
+from trezor import log, loop, protobuf, utils, workflow
+from trezor.enums import FailureType
+from trezor.messages import Failure
+from trezor.wire.context import UnexpectedMessageException, with_context
+from trezor.wire.errors import ActionCancelled, DataError, Error, UnexpectedMessage
+from trezor.wire.protocol_common import Context, Message
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Container
+
+    from trezor.wire import Handler, LoadedMessageType, WireInterface
+
+    HandlerFinder = Callable[[Any, Any], Handler | None]
+    Filter = Callable[[int, Handler], Handler]
+
+# If set to False protobuf messages marked with "experimental_message" option are rejected.
+EXPERIMENTAL_ENABLED = False
+
+
+def wrap_protobuf_load(
+    buffer: bytes,
+    expected_type: type[LoadedMessageType],
+) -> LoadedMessageType:
+    try:
+        if __debug__ and utils.EMULATOR and utils.USE_THP:
+            log.debug(
+                __name__,
+                "Buffer to be parsed to a LoadedMessage: %s",
+                utils.get_bytes_as_str(buffer),
+            )
+        msg = protobuf.decode(buffer, expected_type, EXPERIMENTAL_ENABLED)
+        if __debug__ and utils.EMULATOR:
+            log.debug(
+                __name__, "received message contents:\n%s", utils.dump_protobuf(msg)
+            )
+        return msg
+    except Exception as e:
+        if __debug__:
+            log.exception(__name__, e)
+        if e.args:
+            raise DataError("Failed to decode message: " + " ".join(e.args))
+        else:
+            raise DataError("Failed to decode message")
+
+
+_PROTOBUF_BUFFER_SIZE = const(8192)
+
+WIRE_BUFFER = bytearray(_PROTOBUF_BUFFER_SIZE)
```

Comment:
this file does not _use_ `WIRE_BUFFER` and it's super weird to have this supposedly persistent buffer stowed away in a file that doesn't use it for safe-keeping.

why not keep it in `wire.__init__` where it was before?
(moving it may make sense if you also move `handle_session`)

Reply:
`WIRE_BUFFER` moved to `wire.__init__` here: https://github.com/trezor/trezor-firmware/pull/4357/commits/eda6521d11b2d045c1d77aaa58584f5e927d9271